ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlInternalTopicUtils.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.util;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.exception.KafkaTopicException;
+import io.confluent.ksql.util.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlConstants;
+import org.apache.kafka.common.config.TopicConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KsqlInternalTopicUtils {
+  private static final Logger log = LoggerFactory.getLogger(KsqlInternalTopicUtils.class);
+
+  public static String getTopicName(final KsqlConfig ksqlConfig, final String topicSuffix) {
+    return String.format(
+        "%s%s_%s",
+        KsqlConstants.KSQL_INTERNAL_TOPIC_PREFIX,
+        ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+        topicSuffix
+    );
+  }
+
+  public static void ensureTopic(final String name,
+                                 final KsqlConfig ksqlConfig,
+                                 final KafkaTopicClient topicClient) {
+    ensureTopic(name, ksqlConfig, topicClient, TopicConfig.CLEANUP_POLICY_DELETE);
+  }
+
+  public static void ensureTopic(final String name,
+                                 final KsqlConfig ksqlConfig,
+                                 final KafkaTopicClient topicClient,
+                                 final String cleanupPolicy) {
+    final long requiredTopicRetention = Long.MAX_VALUE;
+
+    if (topicClient.isTopicExists(name)) {
+      final ImmutableMap<String, Object> requiredConfig =
+          ImmutableMap.of(TopicConfig.RETENTION_MS_CONFIG, requiredTopicRetention);
+
+      if (topicClient.addTopicConfig(name, requiredConfig)) {
+        log.info(
+            "Corrected retention.ms on ksql internal topic. topic:{}, retention.ms:{}",
+            name,
+            requiredTopicRetention);
+      }
+
+      return;
+    }
+
+    final short replicationFactor
+        = ksqlConfig.getShort(KsqlConfig.SINK_NUMBER_OF_REPLICAS_PROPERTY);
+    if (replicationFactor < 2) {
+      log.warn("Creating topic {} with replication factor of %d which is less than 2. "
+          + "This is not advisable in a production environment. ", name, replicationFactor);
+    }
+    try {
+      topicClient.createTopic(
+          name,
+          1,
```

Comment:
nit: what is the `1`? number of partitions? might be good to put it into a field/variable